lib/liquid/combinators/tag.ex

```diff
@@ -59,4 +81,32 @@ defmodule Liquid.Combinators.Tag do
     |> ignore(string("end" <> tag_name))
     |> parsec(:end_tag)
   end
+
+  def store_tag_in_context(_rest, [{:block, tag}] = acc, %{tags: tags} = context, _line, _offset) do
+    tag_name = tag |> Keyword.keys() |> hd() |> to_string()
+    {acc, %{context | tags: [tag_name | tags]}}
+  end
+
+  def check_allowed_tags(_, acc, %{tags: []} = context, _, _, _) do
```

Comment:
Could you add `@doc`umentation and `@spec`s?

Reply:
Done